packages/edit-navigation/src/components/menu-editor/index.js

```diff
@@ -12,16 +18,48 @@ import {
 import { __ } from '@wordpress/i18n';
 import { Panel, PanelBody, Button } from '@wordpress/components';
 import { useViewportMatch } from '@wordpress/compose';
+import { useSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import useNavigationBlocks from './use-navigation-blocks';
 import MenuEditorShortcuts from './shortcuts';
 
-export default function MenuEditor( { menuId, blockEditorSettings } ) {
+export default function MenuEditor( {
+	menuId,
+	blockEditorSettings,
+	onDelete,
+} ) {
+	const kind = 'root';
+	const name = 'menu';
+
 	const [ blocks, setBlocks, saveBlocks ] = useNavigationBlocks( menuId );
 	const isLargeViewport = useViewportMatch( 'medium' );
+	const entities = useSelect( ( select ) =>
+		select( 'core' ).getEntitiesByKind( kind )
+	);
+
+	const entity = find( entities, { kind, name } );
+
+	const deleteMenu = async ( recordId ) => {
+		const path = `${ entity.baseURL + '/' + recordId + '?force=true' }`;
+		const deletedRecord = await apiFetch( {
+			path,
+			method: 'DELETE',
+		} );
+		return deletedRecord.previous;
+	};
+
+	const askToDelete = async () => {
+		if (
+			// eslint-disable-next-line no-alert
+			window.confirm( __( 'Are you sure you want to delete this menu?' ) )
+		) {
+			const deletedMenu = await deleteMenu( menuId );
+			onDelete( deletedMenu );
+		}
+	};
```

Comment:
Let's move the delete button, `askToDelete`, and `deleteMenu` into a new `DeleteButton` component so that `MenuEditor` stays relatively focused.